write-fonts/src/tables/gvar.rs

```diff
@@ -0,0 +1,451 @@
+//! The gvar table
+
+include!("../../generated/generated_gvar.rs");
+
+use std::collections::HashMap;
+
+use crate::{collections::HasLen, OffsetMarker};
+
+use super::variations::{
+    PackedDeltas, PackedPointNumbers, Tuple, TupleVariationCount, TupleVariationHeader,
+};
+
+/// Variation data for a single glyph, before it is compiled
+pub struct GlyphVariations {
+    gid: GlyphId,
+    variations: Vec<GlyphDeltas>,
+}
+
+/// Glyph deltas for one point in the design space.
+pub struct GlyphDeltas {
+    peak_tuple: Tuple,
+    intermediate_region: Option<(Tuple, Tuple)>,
+    deltas: Vec<(i16, i16)>,
+}
+
+impl Gvar {
+    /// Construct a gvar table from a vector of per-glyph variations.
+    ///
+    /// Variations must be present for each glyph, but may be empty.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the axis count is not consistent between variations (excluding
```

Comment:
why not an Error?

Reply:
It was an error initially, but I couldn't see any way that would be used by the caller other than unwrapping. My basic thinking, here, is that it is the caller's responsibility to ensure that the invariants are met, and if they aren't then there is a bug in the caller's implementation; there is no normal circumstance where one of these error conditions might be triggered.